### PR TITLE
Add AssertReplyContain

### PR DESCRIPTION
**Pull request by baruchiro**

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs [hunk 1]
```diff
@@ -165,6 +165,36 @@ public TestFlow AssertReply(string expected, string description = null, uint tim
...
+        /// <summary>
+        /// Adds an assertion that the turn processing logic responds as expected.
+        /// </summary>
+        /// <param name="expected">The part of the expected text of a message from the bot.</param>
+        /// <param name="description">A message to send if the actual response is not as expected.</param>
+        /// <param name="timeout">The amount of time in milliseconds within which a response is expected.</param>
+        /// <returns>A new <see cref="TestFlow"/> object that appends this assertion to the modeled exchange.</returns>
+        /// <remarks>This method does not modify the original <see cref="TestFlow"/> object.</remarks>
+        /// <exception cref="Exception">The bot did not respond as expected.</exception>
+        public TestFlow AssertReplyContain(string expected, string description = null, uint timeout = 3000)
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Typo, method name should probably be AssertReplyContain**s**.

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs [hunk 2]
```diff
@@ -165,6 +165,36 @@ public TestFlow AssertReply(string expected, string description = null, uint tim
...
+        {
+            return AssertReply(
+                (reply) =>
+                {
+                    if (!reply.AsMessageActivity().Text.Contains(expected))
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Possible null reference exception.
AsMessageActivity() will return null if the activity is not of ActivityTypes.Message, check the value for null before accessing the text property.

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs [hunk 3]
```diff
@@ -165,6 +165,36 @@ public TestFlow AssertReply(string expected, string description = null, uint tim
...
+                    {
+                        if (description == null)
+                        {
+                            throw new Exception($"Expected:{expected}\nReceived:{reply.AsMessageActivity().Text}");
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Same as above

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 1 above)
Comment (gabog) [inline libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Please add unit tests for this addition, we don't have a TestFlowTests class at the moment (and we would like to start adding more coverage) so go ahead and create one in the Microsoft.Bot.Builder.Tests project under the Adapters folder (needs to be created).

**Comment by gabog** [comment]
Hi @baruchiro, thank you for your contribution, I requested some changes in my review, please check them out and let me know if you have any question.

Also, please provide detailed comments in the PR the next time you submit one, it will help us triage the request faster.

Thanks again.

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 1 above)
Comment (baruchiro) [reply libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Resolved.
https://github.com/microsoft/botbuilder-dotnet/pull/1922/commits/f8a95c0a462b0359af59290fc70fd1548eae3efd

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 2 above)
Comment (baruchiro) [reply libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: This problem also exists in other functions.
Resolved in https://github.com/microsoft/botbuilder-dotnet/pull/1922/commits/55f15b9e9733779ff378e50d193473051172ca21

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 1 above)
Comment (baruchiro) [reply libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: Added tests in https://github.com/microsoft/botbuilder-dotnet/pull/1922/commits/0a3d14ccc374d3f6c7864ef8a0a55e69381b8c3b

**Comment by cleemullins** [comment]
@gabog I believe all of your concerns were addressed. Could you revisit your review?

**Pull request merged**